orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/load_table.py

```diff
@@ -1,56 +1,122 @@
-from hca_orchestration.solids.load_hca.poll_ingest_job import check_data_ingest_job_result, \
-    check_table_ingest_job_result
 from hca_orchestration.support.typing import HcaScratchDatasetName, MetadataType, MetadataTypeFanoutResult
 
 
+@solid(
+    required_resource_keys={
+        "bigquery_service",
+        "target_hca_dataset",
+        "scratch_config",
+        "data_repo_service",
+        "gcs"
+    },
+)
+def load_table_solid(
+    context: AbstractComputeExecutionContext,
+        metadata_fanout_result: MetadataTypeFanoutResult
+) -> Optional[JobId]:
+    """We want our dagster solids as dumb as possible so this is a simple facade
+    that delegates straight to the load_table function"""
+    return load_table(
+        context.resources.scratch_config,
+        metadata_fanout_result.scratch_dataset_name,
+        metadata_fanout_result.metadata_type,
+        metadata_fanout_result.path,
+        context.resources.target_hca_dataset,
+        context.resources.gcs,
+        context.resources.data_repo_service,
+        context.resources.bigquery_service
+    )
+
+
+def load_table(
+        scratch_config: ScratchConfig,
+        scratch_dataset_name: HcaScratchDatasetName,
+        metadata_type: MetadataType,
+        metadata_path: str,
+        target_hca_dataset: TdrDataset,
+        gcs_client: Client,
+        data_repo_service: DataRepoService,
+        bigquery_service: BigQueryService
+) -> Optional[JobId]:
+    source_path = f"{scratch_config.scratch_prefix_name}/{metadata_path}/{metadata_type}/"
+    if not path_has_any_data(scratch_config.scratch_bucket_name, source_path, gcs_client):
+        logging.info(f"No data for metadata type {metadata_type}")
+        return None
+
+    num_new_rows = start_load(
+        scratch_config,
+        scratch_dataset_name,
+        target_hca_dataset,
+        metadata_type,
+        metadata_path,
+        data_repo_service,
+        bigquery_service
+    )
+
+    if num_new_rows == 0:
+        logging.info(f"No new rows for metadata type {metadata_type}")
+        return None
+
+    logging.info(f"New rows for metadata type {metadata_type}, soft deleting outdated versions...")
+    maybe_outdated_job_id: Optional[JobId] = clear_outdated(
+        scratch_config,
+        target_hca_dataset,
+        metadata_type,
+        bigquery_service,
+        data_repo_service,
+        gcs_client
+    )
+    return maybe_outdated_job_id
+
+
 def _diff_hca_table(
         metadata_type: MetadataType,
         metadata_path: str,
         primary_key: str,
         joined_table_name: str,
         scratch_config: ScratchConfig,
-        target_hca_dataset: HcaDataset,
+        target_hca_dataset: TdrDataset,
         scratch_dataset_name: HcaScratchDatasetName,
         bigquery_service: BigQueryService
 
 ) -> None:
-    datarepo_key = f"{primary_key} as datarepo_{primary_key}"
+    datarepo_key = f"{primary_key} as datarepo_{primary_key}, version as datarepo_version"
```

Comment:
This is the actual bugfix; we weren't joining on the version before so the code was interpreting the presence of _any_ version of an entity as a signal to skip any updates.